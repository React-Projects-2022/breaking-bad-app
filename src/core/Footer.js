const Footer = () => (
  <div className="footer">
    <p>Breaking Bad with React - Anartz Mugika Ledo - { new Date().getUTCFullYear()}</p>
  </div>
);

export default Footer;
