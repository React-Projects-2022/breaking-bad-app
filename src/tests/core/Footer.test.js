import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import Footer from "../../core/Footer";
jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
  
}));
describe("Pruebas en Footer", () => {
  const wrapper = shallow(<Footer />);

  test("Debe de mostrar <Footer /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Comprobando el texto del Footer si es correcto", () => {
    const expectText = `Breaking Bad with React - Anartz Mugika Ledo -  ${new Date().getFullYear()}  /   label english spanish basque`;
    expect(wrapper.find("p").text()).toBe(expectText);
  });

  test("Comprobar que se aplica correctamente el estilo al div principal", () => {
    const expectClassName = "footer";
    expect(wrapper.find("div").props().className).toBe(expectClassName);
  });

  test("Cambiar de idioma", () => {
    // wrapper.find("button").at(1).simulate("click");
  })
});
