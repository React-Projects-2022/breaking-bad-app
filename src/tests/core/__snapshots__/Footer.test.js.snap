// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Footer Debe de mostrar <Footer /> correctamente 1`] = `
<div
  className="footer"
>
  <p>
    Breaking Bad with React - Anartz Mugika Ledo - 
    2021
  </p>
</div>
`;
